src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodServiceImpl.java

```diff
@@ -111,10 +135,25 @@ public BreedingMethodDTO edit(final Integer breedingMethodDbId, final BreedingMe
 			}
 		}
 
+		List<GermplasmNameTypeDTO> germplasmNameTypeDTOs = new ArrayList<>();
+		if (StringUtils.isNotEmpty(breedingMethod.getSnameTypeCode())) {
+			germplasmNameTypeDTOs =
+				this.germplasmNameTypeService.filterGermplasmNameTypes(Collections.singleton(breedingMethod.getSnameTypeCode()));
+			method.setSnametype(germplasmNameTypeDTOs.get(0).getId());
+		} else {
+			// Set to null for scenario where the user removed the snametype value
+			method.setSnametype(null);
+		}
+
 		final BreedingMethodMapper mapper = new BreedingMethodMapper();
 		mapper.mapForUpdate(breedingMethod, method);
 		this.daoFactory.getMethodDAO().update(method);
-		return new BreedingMethodDTO(method);
+		final BreedingMethodDTO breedingMethodDTO = new BreedingMethodDTO(method);
```

Comment:
@mderamos I would also add snameTypeId to the response in case we want to redirect to the name type edition for example. 
I would suggest to have id in the request and then name in the DTO